src/main/scala/intellij/haskell/module/HaskellModuleBuilder.scala

```diff
@@ -218,12 +223,18 @@ object HaskellModuleBuilder {
     }
   }
 
-  private def downloadHaskellPackageSources(project: Project, stackPath: String, haskellPackages: Seq[HaskellPackageInfo], progressIndicator: ProgressIndicator) = {
+  private def downloadHaskellPackageSources(project: Project, libRoot: File, stackPath: String, haskellPackages: Seq[HaskellPackageInfo], progressIndicator: ProgressIndicator) = {
     val step = 0.8 / haskellPackages.size
     var progressFraction = InitialProgressStep
-    haskellPackages.flatMap { packageInfo =>
-      val fullName = packageInfo.name + "-" + packageInfo.version
-      val stdErr = CommandLine.runProgram(Some(project), project.getBasePath + File.separator + LibName, stackPath, Seq("unpack", fullName), 10000, captureOutputToLog = true, logErrorAsInfo = true).map(_.getStderr)
+
+    libRoot.listFiles().filterNot(subDir => haskellPackages.map(_.dirName).contains(subDir.getName)).foreach(FileUtil.delete)
+
+    haskellPackages.filter(packageInfo => {
```

Comment:
I think this is not a good approach since we always need to [check](https://github.com/rikvdkleij/intellij-haskell/blob/master/src/main/scala/intellij/haskell/module/HaskellModuleBuilder.scala#L235-L239) whether we can unpack those un-downloaded packages. But I just update the code and give each package list a name to make that code block more intelligible 😄 
>There have to be 2 lists, one for downloading packages which do not exist yet and one for adding to the library table.